lightning-invoice/src/utils.rs

```diff
@@ -688,6 +900,59 @@ mod test {
 		}
 	}
 
+	#[test]
+	#[cfg(feature = "std")]
+	fn create_phantom_invoice_with_description_hash() {
+		let mut chanmon_cfgs = create_chanmon_cfgs(3);
+		let seed_1 = [42 as u8; 32];
+		let seed_2 = [43 as u8; 32];
+		let cross_node_seed = [44 as u8; 32];
+		chanmon_cfgs[1].keys_manager.backing = PhantomKeysManager::new(&seed_1, 43, 44, &cross_node_seed);
+		chanmon_cfgs[2].keys_manager.backing =PhantomKeysManager::new(&seed_2, 43, 44, &cross_node_seed);
+		let node_cfgs = create_node_cfgs(3, &chanmon_cfgs);
+		let node_chanmgrs = create_node_chanmgrs(3, &node_cfgs, &[None, None, None]);
+		let nodes = create_network(3, &node_cfgs, &node_chanmgrs);
+		let chan_0_1 = create_announced_chan_between_nodes_with_value(
+			&nodes, 0, 1, 100000, 10001, InitFeatures::known(), InitFeatures::known(),
+		);
+		nodes[0].node.handle_channel_update(&nodes[1].node.get_our_node_id(), &chan_0_1.1);
+		nodes[1].node.handle_channel_update(&nodes[0].node.get_our_node_id(), &chan_0_1.0);
+		let chan_0_2 = create_announced_chan_between_nodes_with_value(
+			&nodes, 0, 2, 100000, 10001, InitFeatures::known(), InitFeatures::known(),
+		);
+		nodes[0].node.handle_channel_update(&nodes[2].node.get_our_node_id(), &chan_0_2.1);
+		nodes[2].node.handle_channel_update(&nodes[0].node.get_our_node_id(), &chan_0_2.0);
+
+		let payment_amt = 20_000;
+		let (payment_hash, payment_secret) = nodes[1].node.create_inbound_payment(Some(payment_amt), 3600).unwrap();
+		let route_hints = vec![
+			nodes[1].node.get_phantom_route_hints(),
+			nodes[2].node.get_phantom_route_hints(),
+		];
+
+		let description_hash = crate::Sha256(Hash::hash("Description hash phantom invoice".as_bytes()));
+		let invoice = ::utils::create_phantom_invoice_with_description_hash::<
+			EnforcingSigner,
+			&test_utils::TestKeysInterface,
+		>(
+			Some(payment_amt), description_hash, payment_hash, payment_secret, route_hints,
+			&nodes[1].keys_manager, Currency::BitcoinTestnet,
+		)
+		.unwrap();
```

Comment:
Why did you `\n` here?

Reply:
No particular reason. Removed with 24ed8af